pkg/mcs/exportspec_test.go

```diff
@@ -40,38 +41,100 @@ func TestMarshalUnmarshal(t *testing.T) {
 		expected *mcs.ExportSpec
 		mutate   func(mcs.ExportSpec) *mcs.ExportSpec
 	}{
-		/*
-			"empty": {},
-			"missing: timestamp": {},
-			"missing: cluster": {},
-			"missing: namespace": {},
-			"missing: name": {},
-			"missing: service":       {},
-			"missing: service.type":  {},
-			"missing: service.ports": {},
-			"simple spec": {},
-			"multiple ports":         {},
-		*/
+
+		"empty": {
+			expected: &mcs.ExportSpec{},
+			mutate:   func(mcs.ExportSpec) *mcs.ExportSpec { return &mcs.ExportSpec{} },
+		},
+
+		"missing: timestamp": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.CreatedAt = metav1.Unix(0, 0)
+				return &es
+			},
+		},
+
+		"missing: cluster": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.ClusterID = ""
+				return &es
+			},
+		},
+
+		"missing: namespace": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Namespace = ""
+				return &es
+			},
+		},
+
+		"missing: name": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Name = ""
+				return &es
+			},
+		},
+
+		"missing: service": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service = mcs.GlobalProperties{}
+				return &es
+			},
+		},
+
+		"missing: service.type": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Type = ""
+				return &es
+			},
+		},
+
+		"missing: service.ports": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				es.Service.Ports = []mcsv1a1.ServicePort{}
+				return &es
+			},
+		},
+
+		"simple spec": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				return &es
+			},
+		},
+
+		"multiple ports": {
+			mutate: func(es mcs.ExportSpec) *mcs.ExportSpec {
+				new_port_81 := mcsv1a1.ServicePort{Port: 81, Name: "http", Protocol: corev1.ProtocolTCP}
+				new_port_82 := mcsv1a1.ServicePort{Port: 82, Name: "http", Protocol: corev1.ProtocolTCP}
+				es.Service.Ports = append(es.Service.Ports, new_port_81)
+				es.Service.Ports = append(es.Service.Ports, new_port_82)
+				return &es
+			},
+		},
 	}
 
 	assertions := require.New(t)
 
 	for name, test := range testcases {
 		t.Logf("Running test case %s", name)
-
 		md := &metav1.ObjectMeta{}
 		test.expected = test.mutate(sample)
-		err := sample.MarshalObjectMeta(md)
+		err := test.expected.MarshalObjectMeta(md)
 		assertions.NoError(err)
-
 		actual := &mcs.ExportSpec{}
 		err = actual.UnmarshalObjectMeta(md)
 		assertions.NoError(err)
-
+		convertTimesToUTC(&test.expected.CreatedAt, &actual.CreatedAt)
 		assertions.Equal(test.expected, actual)
 	}
 }
 
+func convertTimesToUTC(t1 *metav1.Time, t2 *metav1.Time) {
+	loc, _ := time.LoadLocation("UTC")
```

Comment:
can this be redone as:
```go
   t1 := t1.Time.Truncate(time.Second).UTC()
   t1 := t2.Time.Truncate(time.Second).UTC()
```
?
If so, I would just inline this code into the call site and not make it a function call

Reply:
Yes, it possible, I changed edited this as inline with those lines:

```golang
    test.expected.CreatedAt.Time = test.expected.CreatedAt.UTC()
    actual.CreatedAt.Time = actual.CreatedAt.Time.UTC()
```

(Without the Truncate to seconds since we already did it in the sample initialization, so it's an additional change that simplify a bit the code).
If you prefer that the Truncate will stay so it will be clear that it Truncate before comparing (or for future samples) I can keep it.